docs/arch/adr-59.md

```diff
@@ -0,0 +1,19 @@
+# ADR 58: Incremental Static Regeneration for project pages
+
+## Status
+Accepted
+
+## Context
+When the first Engaging Crowds projects launched, project pages were built dynamically with `getServerSideProps`. Page builds were slow, with pages sometimes taking several seconds to load in the browser ([#2741](https://github.com/zooniverse/front-end-monorepo/issues/2741).)
+
+## Decision
+Use [Incremental Static Regeneration](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration) (ISR) to build project pages. ISR is a feature of Next.js, which allows page HTML and JSON to be built and cached on demand, then served with `stale-while-revalidate` cache contol headers. While a page is building, visitors will get stale, cached content, resulting in a faster experience.
+
+## Consequences
+- Page builds were switched from `getServerSideProps` to `getStaticProps`, with a revalidation time of 60s. Revalidation can be set individually for each page route.
+- `getStaticProps` can still be slow, when it runs, so data-fetching from the Panoptes API may need further optimisation ([#3341](https://github.com/zooniverse/front-end-monorepo/issues/3341).)
+- Production projects and staging projects each have their own static routes: `/production/:owner/:project` and `/staging/:owner/:project`. Next.js middleware rewrites incoming request URLs to the production and staging routes used by the project app eg. https://fe-project.preview.zooniverse.org/projects/production/nora-dot-eisner/planet-hunters-tess
```

Comment:
In middleware.js, I see `.rewrite()` when an incoming request URL has a locale param, but there's also rules using `.next()` instead of redirect:
https://github.com/zooniverse/front-end-monorepo/blob/54bd14af4c919b5f3c046f96a905deb6e88b2232/packages/app-project/middleware.js#L49-L66

Would you be willing to add a short comment in middleware.js that explains the difference between`.next()` and `redirect()` in relation to your point here about rewriting to production and staging routes?

Reply:
Sure. `Response.next()` carries on routing without modifying the response. Here, it’s called when a URL doesn’t need rewriting. 
https://nextjs.org/docs/app/api-reference/functions/next-response#next

Reply:
This is an example of a link that won’t be rewritten:
https://fe-project.preview.zooniverse.org/projects/production/nora-dot-eisner/planet-hunters-tess

Reply:
Looking at the middleware again, I might rearrange it, so that all the responses that should not be rewritten are at the top, followed by the responses that are rewritten.

Reply:
I've rewritten the item about rewrites, rearranged the middleware, and added more comments to `middleware.js`.

Reply:
Perfect - thank you!